### ISSUE TITLE
tripal 3 entity chado  FK fields  are numerical input instead of dropdown

### ISSUE THREAD

**Issue by bradfordcondon** [tripal/tripal#210]
As an example, the phylogenetic trees bundle has a numerical input box for analysis_id.  But i think this happens for all auto-generated bundles.  The protocols module arraydesign table, for example, has a column for protocol_id where the widget is a raw input box.

Clearly it would be better to have a dropdown of analyses, protocols, etc, so do we need to create a field for each individually?

![screen shot 2017-12-06 at 5 29 38 pm](https://user-images.githubusercontent.com/7063154/33688891-37e52198-daab-11e7-9afe-9388ed8c6028.png)

**Comment by spficklin** [comment]
Hi @bradfordcondon. Yes this occurs when we have no field other than the default one that Tripal makes. For example, when a new content type is mapped to  a Chado table, Tripal will create a Drupal field for every single column of that table, but by default these fields are just simple Drupal field types.  If a a field needs something more than the Drupal default then we need a special TripalField for it.  This is how the feature.organism_id and stock.organism_id columns are supported.  The **obi__organism** field takes care of those columns.  You don't see a textfield expecting an integer for those fields.  They have organism drop-downs instead.   

We need to create the fields that will support analyses and protocols.   These should be fairly easy to create and we can and should certainly add that to Tripal v3 for the next release.

**Comment by bradfordcondon** [comment]
OK.   So we do require field for each.

I can contribute these fields to the main tripal module and use the `obi__organism` field as as reference.  Unless you all are already working on this?

Cheers

**Comment by bradfordcondon** [comment]
Took a stab at Protocol #213

**Comment by bradfordcondon** [comment]
- [ ] Analysis FK field (figure out #217) first
- [x] Protocol FK  field #254 

I'll track additional FK fields as necessary...

**Comment by spficklin** [comment]
Just checking on this. @bradfordcondon do you want to contribute your fields to the tripal_chado module? Are they sufficently generic?

**Comment by bradfordcondon** [comment]
what happened was i needed help understanding the field methods related to searching/sorting and you said wait for the documentation. 

I'm sure if I revisit this I'd find that the fields all need work...

**Comment by bradfordcondon** [comment]
Specifically:

>But the field is missing an elementInfo(), query() and queryOrder() functions. Do you think you could add those and add another pull request?

**Comment by laceysanderson** [comment]
So then this is currently blocked by #266.

**Comment by bradfordcondon** [comment]
Added a PR with the requested methods.

RE: analysis... I think I only have our organism_analysis linker field (local__analysis_browser) https://github.com/statonlab/tripal_manage_analyses/tree/master/includes/TripalFields.  This isnt the same as a linker field...

I guess we still need an analysis linker field.  My Protocol field was super minimal with just the name and hte type.  For analysis it would probably just be the name?

**Comment by spficklin** [comment]
I added a PR to fix this.  I think the analysis field is the last FK field that needed a field for our current list of content types.  Others were added by PR #398.  So aftwerwards this new PR gets merged we can close this issue.

**Comment by bradfordcondon** [comment]
merged.  Is that really all the FK fields?  great.  closing.
